src/components/MoneyRequestAmountInput.tsx

```diff
@@ -223,7 +247,7 @@ function MoneyRequestAmountInput(
 
         // we want to re-initialize the state only when the amount changes
         // eslint-disable-next-line react-compiler/react-compiler, react-hooks/exhaustive-deps
-    }, [amount, shouldKeepUserInput]);
+    }, [amount, formatAmountOnBlur, shouldKeepUserInput, onFormatAmount, shouldResetAmount, onResetAmount]);
```

Comment:
I think I'm missing something, could you elaborate on why do we need those dependencies?
`onResetAmount` makes sense, but I'm confused about `formatAmountOnBlur`, `onFormatAmount` and `shouldResetAmount`

Reply:
> I'm getting the following warning 

@kaushiktd, is this an IDE warning, warning when running linter locally or does the GH lint check fails?

`// eslint-disable-next-line react-compiler/react-compiler, react-hooks/exhaustive-deps` should disable this warning (at least it does for me when running linter locally)

Reply:
@eVoloshchak  this is GH lint check fails. how i disable it

Reply:
@eVoloshchak  this is GH lint check fails. how i disable it

Reply:
@kaushiktd, we shouldn't disable it, but it also shouldn't be different from running the liner check locally. 
Could you push the changes without the additional dependencies? I'd take a look at why GH lint and `npm run lint` yield different results